examples/bri-3/src/bri/identity/bpiSubjects/models/bpiSubject.ts

```diff
@@ -41,4 +40,16 @@ export class BpiSubject {
   public get publicKey(): string {
     return this._publicKey;
   }
+
+  public updateName(newName: string): void {
```

Comment:
not hard opinion on this, but why not make name, description and publickkey public fields? it would eliminate 6 methods and will have same behavior

Reply:
Because this is just an example. Usually you will update more than one field as part of the transaction (i.e. UpdateGeneralInfomation). Updating the fields will be accompanied by a set of business rules, and we want to avoid the possibility that somewhere you can introduce a change directly and elsewhere you do it through a method. And we are good OO pupils ;).